Test the FPGA CI Test Shield on all targets - DO NOT MERGE

c1728p9 opened the pull request:
### Description

Build in the FPGA CI Test Shield files to all targets so CI can ensure there are no compatibility issues.

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

c1728p9 committed:
Test the FPGA CI Test Shield on all targets

Build in the FPGA CI Test Shield files to all targets so CI can ensure
there are no compatibility issues.

fkjagodzinski commented on targets/targets.json:
```diff
@@ -15,6 +15,7 @@
         "default_lib": "std",
         "bootloader_supported": false,
         "static_memory_defines": true,
+        "components_add": ["FPGA_CI_TEST_SHIELD"],
```
```suggestion
```
I think we should move this to `"components": ["PSA_SRV_IMPL", "PSA_SRV_EMUL", "NSPE", "FPGA_CI_TEST_SHIELD"],` above.

mprse replied:
Nice catch! Probably we need new PR.

0xc0170 committed:
targets: fix components addition

Co-Authored-By: Filip Jagodziński <fkjagodzinski@gmail.com>

fkjagodzinski committed:
Test the FPGA CI Test Shield on all targets

Build in the FPGA CI Test Shield files to all targets so CI can ensure
there are no compatibility issues.

Based on: 950e14031f07f91823ea611f137996961013bd24
